src/scripts/choices.js

```diff
@@ -558,7 +558,9 @@ class Choices {
       }
 
       // it's a choices fetcher
-      requestAnimationFrame(() => this._handleLoadingState(true));
+      ({ then: requestAnimationFrame.bind(window) }.then(() =>
+        this._handleLoadingState(true),
+      ));
```

Comment:
This is not looking pretty. I think the whole problem here is not to wait for `_handleLoadingState` callback return. So, whole fragment can be rewritten like this:

```js
      // it's a choices fetcher
      const fetcher = choicesArrayOrFetcher(this);
      if (typeof Promise === 'function' && fetcher instanceof Promise) {
        // that's a promise
        // eslint-disable-next-line compat/compat
        return new Promise(requestAnimationFrame.bind(window))
          .then(() => this._handleLoadingState(true))
          .then(() => fetcher)
          .then(data => this.setChoices(data, value, label, replaceChoices))
          .catch(err => {
            if (!this.config.silent) {
              console.error(err);
            }
          })
          .then(() => this._handleLoadingState(false))
          .then(() => this);
      }
```

Reply:
I think trying to make `requestAnimationFrame` behave like a promise is just going to confuse people to be honest.

Reply:
@jshjohnson do you mean to write it completely instead?
```js
new Promise(resolve => requestAnimateFrame(resolve))
```
it's just transpiles to an even longer version, so, `bind` was handy here... _Confused_ developer will not touch this line - that what we want, doesn't it?

Reply:
`new Promise(resolve => requestAnimateFrame(resolve))` is much more readable to `new Promise(requestAnimationFrame.bind(window)` in my opinion

Reply:
@tinovyatkin @jshjohnson  I was trying with `new Promise()` but your prettier does not accept that. Please, implement it the best way you think.